network_simulator/plot_tool.py

```diff
@@ -0,0 +1,390 @@
+        (packet size) / (time interval).
+        """
+        curr_bit_sum = 0
+        curr_window_end_time = GRAPH_WINDOW_SIZE
+        output = []
+        for tup_idx in range(len(tuple_list)):
+            timestamp, packet_size = tuple_list[tup_idx]
+            if timestamp > curr_window_end_time:
+                output.append(
+                    (curr_window_end_time,
+                     float(curr_bit_sum) / GRAPH_WINDOW_SIZE / MEGABIT))
+                curr_bit_sum = 0
+                curr_window_end_time += GRAPH_WINDOW_SIZE
+            if timestamp <= curr_window_end_time:
+                curr_bit_sum += packet_size
+        # Append the last elements if they did not reach an end of an
+        # interval.
+        if curr_bit_sum != 0:
+            output.append(
+                (curr_window_end_time,
+                 float(curr_bit_sum) / GRAPH_WINDOW_SIZE / MEGABIT))
+        return output
+
+    @staticmethod
+    def graph_tuple_list(tuple_list=[], scatter=False):
+        """
+        Graphs the content of a tuple list.
+
+        :param list tuple_list: current input tuple list to be plotted
+        (this list is a 2-D list containing (timestamp, data) tuples).
+        :param boolean scatter: if we want a scatter plot, we set this
+        variable to True; otherwise, it would be a line graph.
+        """
+        if not tuple_list:
+            return
+
+        if scatter:
+            # Scatter plot.
+            plt.scatter(*zip(*tuple_list), marker='o', linestyle='--')
+        else:
+            # Line graph
+            plt.plot(*zip(*tuple_list), marker='o', linestyle='--')
+
+    @staticmethod
+    def output_sum_avg_tuple_list(id, value_type, count, units, tuple_list=[]):
+        """
+        Outputs the average value of a tuple list. This is a simple
+        (sum / count) or (sum / time).
+
+        Assume that a tuple list is in the format of (timestamp, data), we
+        iterature through the data field of the tuples and outputs the
+        average value to std out.
+
+        :param string id: the Device of this average.
+        :param string value_type: the data type of the average.
+        :param boolean count: whether the denominator of the average is a simple
+        count (True), or a timestamp value (False).
+        :param string units: units of the average. e.g. "(sec)"
+        :param list tuple_list: the tuple list to be computed.
+        :return: no return, just print out the statement to stdout.
+        """
+        curr_sum = 0
+        for time, data in tuple_list:
+            curr_sum += data
+        avg = 0
+        if len(tuple_list) != 0:
+            if count:
+                avg = float(curr_sum) / len(tuple_list)
+            else:
+                latest_timestamp = tuple_list[len(tuple_list) - 1][0]
+                avg = float(curr_sum) / latest_timestamp
+        print id + " Average " + value_type + ": " + str(avg) + " " + units
```

Comment:
You should import logging and set "logger = logging.getLogger(__name__)" at the top of the file (e.g. see network_topology.py). Then, use logger.info() wherever possible, instead of "print". logger.info can work with strings as you've passed in, or with format strings:

logger.info("%s Average %s: %f %s", id, value_type, avg, units)

or something like that.